core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -902,7 +951,15 @@ void CudaSpaceInitializer::finalize(bool all_spaces) {
   }
 }
 
-void CudaSpaceInitializer::fence() { Kokkos::Cuda::impl_static_fence(); }
+void CudaSpaceInitializer::fence() {
+  Kokkos::Cuda::impl_static_fence(
+      "Kokkos::CudaSpaceInitializer::fence: Initializer Fence");
+}
+void CudaSpaceInitializer::fence(const std::string &name) {
+  // Kokkos::Cuda::impl_static_fence("Kokkos::CudaSpaceInitializer::fence:
+  // "+name); //TODO: or this
+  Kokkos::Cuda::impl_static_fence(name);
```

Comment:
Is this something you want to have discussed and addressed in this pull request?

Reply:
I'll delete the comment. I basically decided on "do it the way it is now." This fence is never directly invoked, only ever invoked by another function, that function should be saying what the name of the fence is